### PR TITLE
[search] Viewport filtering fix.

### PR DESCRIPTION
**Pull request by vng**
Closes https://github.com/organicmaps/organicmaps/issues/3680

### CODE REVIEW

File: search/engine.cpp [hunk 1]
```diff
@@ -8,19 +8,16 @@
...
-#include <cstdint>
-#include <functional>
 #include <map>
 #include <vector>
 
-using namespace std;
-
 namespace search
 {
+using namespace std;
```
Comment (biodranik) [inline search/engine.cpp:19]: Давай не вставлять юзинг всего неймспейса нигде. Только конкретных классов.

**Review by biodranik — APPROVED** [review]

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 13

Tested with fire_hydrant and defibrillator

Is it possible to improve speed to show blue indicator in the futur ? (Load blue indicator in map is to long when you look map after search and when you change scale).

**Comment by biodranik** [comment]
> Is it possible to improve speed to show blue indicator in the futur ? (Load blue indicator in map is to long when you look map after search and when you change scale).

@Jean-BaptisteC blue highlighted search results appear when our search finds features. You don't use an SD card, right? (sometimes they are very slow).

@vng We can also improve visual UX for users by _immediately_ displaying _any_ found to result in the viewport, without waiting until some batch will be collected, or some timeout will finish. Batching is important for _sorting_ results in the list, it is completely irrelevant to results on the map. The only issue we need to keep in mind is a properly visible search indicator that will clearly say to users that the search has not finished yet. And of course, we need some way to let users know if _nothing_ was found in the viewport. Maybe some toast message like "%s was not found, try to zoom out"?
It may worth to create a separate issue for that :)

File: search/engine.cpp (hunk 1 above)
Comment (vng) [reply search/engine.cpp:19]: Я тут переместил существующий.

**Comment by vng** [comment]
@Jean-BaptisteC Is it slow on Pixel 6? Could you please make a short video? Do you search above Metz like in the issue?

**Comment by Jean-BaptisteC** [comment]
I don't use SD Card.
If i reduce scale and increase scale after (scale 15, scale 7 to scale 15), reload result is long to show on map

https://user-images.githubusercontent.com/87148630/197404907-4b785875-83a8-46f6-a381-dd18f07ecfa2.mp4

**Comment by vng** [comment]
Ah, you built Debug version. There is no reason to check Debug for speed. Please, build Release.

**Comment by Jean-BaptisteC** [comment]
I can't build Release version (I have too bad connection to download repo).
The real problem is fixed all fire_hydrants arround me is show in map.
Thanks for correction :)

File: search/engine.cpp (hunk 1 above)
Comment (biodranik) [reply search/engine.cpp:19]: Лучше сразу выпили, unity не будет падать )

**Pull request merged**